### ISSUE TITLE
Servlet tests depend on invalid default charset of ISO-8859-1

### ISSUE THREAD

**Issue by joakime** [jakartaee/platform-tck#589]
The tests in ...

https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/servlet/spec/i18n/encoding/TestServlet.java

The ISO-8859-1 charset for `text/html` was only defined as a default charset for RFC2616 (and only that specific rfc).

The updated RFC:

https://tools.ietf.org/html/rfc7231#appendix-B - Appendix B.  Changes from RFC 2616

```
   The default charset of ISO-8859-1 for text media types has been
   removed; the default is now whatever the media type definition says.
   Likewise, special treatment of ISO-8859-1 has been removed from the
   Accept-Charset header field.  (Section 3.1.1.3 and Section 5.3.3)
```

Looking at the IANA registered media-types ...

https://www.iana.org/assignments/media-types/media-types.xhtml#text

We see that `text/html` has the following registered - https://www.iana.org/assignments/media-types/text/html
Which doesn't declare a `text/html` specific default, instead leaning on the `text/*` defaults.
Which is defined in https://tools.ietf.org/html/rfc6657 as US-ASCII.

With HTTP/2 and HTTP/3, they defer to Section 3.3 of RFC7230 for the definition of "payload body". (which has no default charset declared for any media-type)

In short, there is no current default charset for `text/html`, there was one briefly defined during the RFC2616, but removed in RFC7231.

**Comment by scottmarlow** [comment]
Hi @joakime 

IMO, this sounds like an issue for the https://github.com/eclipse-ee4j/servlet-api project.  If as part of that discussion, a pull request is then created against https://github.com/eclipse-ee4j/jakartaee-tck, the pull request should mention the corresponding https://github.com/eclipse-ee4j/servlet-api issue.  I would also look for the servlet-api project leads to agree with the jakartaee-tck pull request.

Could you please create a https://github.com/eclipse-ee4j/servlet-api issue for this and mention this issue there (to link the issues) and then we can close this issue (effectively moving this issue to the servlet-api project.

Sorry that this is so complicated. :-)

**Comment by scottmarlow** [comment]
@joakime I started making a change to my previous comment last night but must of forgotten to save it.  I'm going to delete it and respond again here.

Could you please indicate if this issue is simply a servlet test improvement issue or if you are challenging the mentioned test for a particular servlet implementation?  Challenging a TCK test is a different process than simple improving the TCK.

Thank you very much for raising this issue!

Scott
